File: CS/GridMVVMItemsSource/ColumnTemplateSelector.cs
using System.Windows.Controls;
using System.Windows;
using Model;
using DevExpress.Xpf.Core.Native;

namespace View {
    public class ColumnTemplateSelector : DataTemplateSelector {
        public override DataTemplate SelectTemplate(object item, DependencyObject container) {
            Column column = (Column)item;
#if SILVERLIGHT
            return (DataTemplate)LayoutHelper.FindParentObject<UserControl>(container).Resources[column.Settings + "ColumnTemplate"];
#else
            return (DataTemplate)((Control)container).FindResource(column.Settings + "ColumnTemplate");
#endif
        }
    }
}

File: VB/GridMVVMItemsSource/ColumnTemplateSelector.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Controls
Imports System.Windows
Imports Model
Imports DevExpress.Xpf.Core.Native

Namespace View
	Public Class ColumnTemplateSelector
		Inherits DataTemplateSelector
		Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
			Dim column As Column = CType(item, Column)
#If SILVERLIGHT Then
			Return CType(LayoutHelper.FindParentObject(Of UserControl)(container).Resources(column.Settings & "ColumnTemplate"), DataTemplate)
#Else
			Return CType((CType(container, Control)).FindResource(column.Settings & "ColumnTemplate"), DataTemplate)
#End If
		End Function
	End Class
End Namespace